### ISSUE TITLE
Request for rule enhancement: UnconditionalStringOperationOnLogArgument not just String operations

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#44]
Enhance rule with calling an operation, examples:
bad:
```
LOG.debug("Complete Soap response: {}", getSoapMsgAsString(context.getMsg())); // bad
```
good:
```
if (LOG.isDebugEnabled()) { // good
    LOG.debug("Complete Soap response: {}", getSoapMsgAsString(context.getMsg()));
}
```
To avoid too many false positives, we require one or more arguments for the operation. Also we exclude simple obj.get(arg) calls, like list.get(0) and map.get(key) which are typically fast.

Rule should be renamed to 'UnconditionalOperationOnLogArgument' ('String' removed.)

**Comment by jborgers** [comment]
Done
